hydromt/models/components/grid.py

```diff
@@ -684,98 +411,362 @@ def setup_grid(
 
         # Add region and grid to model
         self.set_geoms(geom, "region")
-        self.set_grid(grid)
+        self.set(grid)
 
         return grid
 
-    ## I/O
-    def read(
-        self,
-        components: List = None,
-    ) -> None:
-        """Read the complete model schematization and configuration from model files.
-
-        Parameters
-        ----------
-        components : List, optional
-            List of model components to read, each should have an associated
-            read_<component> method. By default ['config', 'maps', 'grid',
-            'geoms', 'tables', 'forcing', 'states', 'results']
-        """
-        components = components or [
-            "config",
-            "grid",
-            "geoms",
-            "tables",
-            "forcing",
-            "states",
-            "results",
-        ]
-        super().read(components=components)
-
-    def write(
-        self,
-        components: List = None,
-    ) -> None:
-        """Write the complete model schematization and configuration to model files.
-
-        Parameters
-        ----------
-        components : List, optional
-            List of model components to write, each should have an
-            associated write_<component> method. By default
-            ['config', 'maps', 'grid', 'geoms', 'tables', 'forcing', 'states']
-        """
-        components = components or [
-            "config",
-            "maps",
-            "grid",
-            "geoms",
-            "tables",
-            "forcing",
-            "states",
-        ]
-        super().write(components=components)
-
-    # Properties for subclass GridModel
     @property
-    def res(self) -> Tuple[float, float]:
+    def res(self) -> Tuple[float, float] | None:
         """Returns the resolution of the model grid."""
-        if len(self.grid) > 0:
-            return self.grid.raster.res
+        if len(self.data) > 0:
```

Comment:
cant `self.data` be None here?

Reply:
No it will be an empty xarray dataset if there is no data